app/services/foreman/plugin.rb

```diff
@@ -93,6 +93,17 @@ def installed?(id)
 
     def initialize(id)
       @id = id.to_sym
+      add_loggers
+    end
+
+    def add_loggers
+      return unless (defined?(Foreman::Logging) && SETTINGS[@id])
+
+      if (loggers = SETTINGS[@id][:loggers])
```

Comment:
Not rambly at all, its good feedback to consider.

For the defaults, I am not able to think what defaults a plugin would need to worry about out of the gate. If the plugin is using things like 'Rails.logger' then they will continue to log to the same place and in the same way as before this patch. The log messages will show up as 'app' in the log files. 

If we want to, for the purpose of making logs easier to read, start to enforce per-plugin logs then we would need some default log creation if not specified within the settings file for the plugin. This could be based on the Rails engine name for consistency across the board. Note that this would only apply to specifically inserted log messages (Rails will continue to log requests and such under the 'app' logger). For example,

```
Katello.logger = ::Logging.logger['katello']
Katello.logger.debug "Debugging message for this katello action"
```

Which would then appear within the logs as:

```
2015-05-13 13:28:22 [katello] [D] Debugging message for this Katello action
```

I think splitting the config and the definition of the loggers themselves makes sense across settings and logging. I would expect then, that the 'environmental' configurations would also live under settings (https://github.com/theforeman/foreman/pull/2376/files#diff-d71db24719150ef52a8afd6ef934f9b4R54)?